lib/Controller/RoomController.php

```diff
@@ -1644,4 +1654,69 @@ protected function changeParticipantType(?int $attendeeId, ?string $userId, ?str
 
 		return new DataResponse();
 	}
+
+	/**
+	 * @NoAdminRequired
+	 * @RequireModeratorParticipant
+	 *
+	 * @param int $state
+	 * @param int|null $timer
+	 * @return DataResponse
+	 */
+	public function setLobby(int $state, ?int $timer = null): DataResponse {
+		$timerDateTime = null;
+		if ($timer !== null && $timer > 0) {
+			try {
+				$timerDateTime = $this->timeFactory->getDateTime('@' . $timer);
+				$timerDateTime->setTimezone(new \DateTimeZone('UTC'));
+			} catch (\Exception $e) {
+				return new DataResponse([], Http::STATUS_BAD_REQUEST);
+			}
+		}
+
+		if (!$this->room->setLobby($state, $timerDateTime)) {
+			return new DataResponse([], Http::STATUS_BAD_REQUEST);
+		}
+
+		if ($state === Webinary::LOBBY_NON_MODERATORS) {
+			$participants = $this->participantService->getParticipantsInCall($this->room);
+			foreach ($participants as $participant) {
+				if ($participant->hasModeratorPermissions()) {
+					continue;
+				}
+
+				$this->participantService->changeInCall($this->room, $participant, Participant::FLAG_DISCONNECTED);
+			}
+		}
+
+		return new DataResponse($this->formatRoomV2andV3($this->room, $this->participant));
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @RequireModeratorParticipant
+	 *
+	 * @param int $state
+	 * @return DataResponse
+	 */
+	public function setSIPEnabled(int $state): DataResponse {
+		$user = $this->userManager->get($this->userId);
+		if (!$user instanceof IUser) {
+			return new DataResponse([], Http::STATUS_UNAUTHORIZED);
+		}
+
+		if (!$this->talkConfig->canUserEnableSIP($user)) {
+			return new DataResponse([], Http::STATUS_UNAUTHORIZED);
```

Comment:
should this be "forbidden" instead since unauthorized would return a 401 and make some clients think that the user is not authenticated (which is the case for the previous if statement).

Reply:
right